### PR TITLE
[Dy2stat] Add Resnet Test for V2 APIs

### PR DESCRIPTION
**Pull request by zhhsplendid**
### PR types

Others
### PR changes

Others
### Describe

Add Resnet Test for V2 APIs.

We used tool from https://www.paddlepaddle.org.cn/documentation/docs/zh/2.0-beta/guides/migration_cn.html#paddle1-xpaddle2-0beta to do v1.x to v2 transformation.

This PR pulled changes from #27430, please DO NOT merge before #27430 is merged

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 1]
```diff
@@ -0,0 +1,362 @@
...
+# NOTE: Reduce batch_size from 8 to 2 to avoid unittest timeout.
+batch_size = 2
+epoch_num = 1
+place = paddle.CUDAPlace(0) if paddle.fluid.is_compiled_with_cuda() \
+    else paddle.CPUPlace()
+MODEL_SAVE_PATH = "./resnet_v2.inference.model"
+DY_STATE_DICT_SAVE_PATH = "./resnet_v2.dygraph"
+program_translator = paddle.jit.ProgramTranslator()
+
+if paddle.fluid.is_compiled_with_cuda():
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.is_compiled_with_cuda`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    paddle.fluid.set_flags({'FLAGS_cudnn_deterministic': True})
+
+
+def optimizer_setting(parameter_list=None):
+    optimizer = paddle.optimizer.Momentum(
+        learning_rate=base_lr,
+        momentum=momentum_rate,
+        weight_decay=paddle.fluid.regularizer.L2Decay(l2_decay),
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.regularizer.L2Decay`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+        conv2 = self.conv2(conv1)
+
+        if self.shortcut:
+            short = inputs
+        else:
+            short = self.short(inputs)
+
+        y = paddle.add(x=short, y=conv2)
+
+        layer_helper = paddle.fluid.layer_helper.LayerHelper(
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:127]: Mark `paddle.fluid.layer_helper.LayerHelper`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+        elif layers == 101:
+            depth = [3, 4, 23, 3]
+        elif layers == 152:
+            depth = [3, 8, 36, 3]
+        num_channels = [64, 256, 512, 1024]
+        num_filters = [64, 128, 256, 512]
+
+        self.conv = ConvBNLayer(
+            num_channels=3, num_filters=64, filter_size=7, stride=2, act='relu')
+        self.pool2d_max = paddle.fluid.dygraph.nn.Pool2D(
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.dygraph.nn.Pool2D`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 5]
```diff
@@ -0,0 +1,362 @@
...
+                initializer=paddle.nn.initializer.Uniform(-stdv, stdv)))
+
+    @paddle.fluid.dygraph.declarative
+    def forward(self, inputs):
+        y = self.conv(inputs)
+        y = self.pool2d_max(y)
+        for bottleneck_block in self.bottleneck_block_list:
+            y = bottleneck_block(y)
+        y = self.pool2d_avg(y)
+        y = paddle.fluid.layers.reshape(y, shape=[-1, self.pool2d_avg_output])
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.layers.reshape`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+        total_acc1 = 0.0
+        total_acc5 = 0.0
+        total_sample = 0
+
+        for batch_id, data in enumerate(data_loader()):
+            start_time = time.time()
+            img, label = data
+
+            pred = resnet(img)
+            loss = paddle.fluid.layers.cross_entropy(input=pred, label=label)
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.layers.cross_entropy`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 7]
```diff
@@ -0,0 +1,362 @@
...
+            avg_loss = paddle.mean(x=loss)
+            acc_top1 = paddle.metric.accuracy(input=pred, label=label, k=1)
+            acc_top5 = paddle.metric.accuracy(input=pred, label=label, k=5)
+
+            avg_loss.backward()
+            optimizer.minimize(avg_loss)
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:243]: Mark `optimizer.minimize` --> `optimizer.step()`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 8]
```diff
@@ -0,0 +1,362 @@
...
+            total_sample += 1
+
+            end_time = time.time()
+            if batch_id % 2 == 0:
+                print( "epoch %d | batch step %d, loss %0.3f, acc1 %0.3f, acc5 %0.3f, time %f" % \
+                    ( epoch, batch_id, total_loss.numpy() / total_sample, \
+                        total_acc1.numpy() / total_sample, total_acc5.numpy() / total_sample, end_time-start_time))
+            if batch_id == 10:
+                if to_static:
+                    paddle.fluid.dygraph.jit.save(resnet, MODEL_SAVE_PATH)
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.dygraph.jit.save`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 9]
```diff
@@ -0,0 +1,362 @@
...
+                else:
+                    paddle.fluid.dygraph.save_dygraph(resnet.state_dict(),
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:260]: Mark `paddle.fluid.dygraph.save_dygraph`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 10]
```diff
@@ -0,0 +1,362 @@
...
+
+    return total_loss.numpy()
+
+
+def predict_dygraph(data):
+    program_translator.enable(False)
+    paddle.disable_static(place)
+    resnet = ResNet()
+
+    model_dict, _ = paddle.fluid.dygraph.load_dygraph(DY_STATE_DICT_SAVE_PATH)
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:275]: Mark `paddle.fluid.dygraph.load_dygraph`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 11]
```diff
@@ -0,0 +1,362 @@
...
+    pred_res = exe.run(inference_program,
+                       feed={feed_target_names[0]: data},
+                       fetch_list=fetch_targets)
+
+    return pred_res[0]
+
+
+def predict_dygraph_jit(data):
+    paddle.disable_static(place)
+    resnet = paddle.fluid.dygraph.jit.load(MODEL_SAVE_PATH)
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Mark `paddle.fluid.dygraph.jit.load`

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py [hunk 12]
```diff
@@ -0,0 +1,362 @@
...
+        static_loss = self.train(to_static=True)
+        dygraph_loss = self.train(to_static=False)
+        self.assertTrue(
+            np.allclose(static_loss, dygraph_loss),
+            msg="static_loss: {} \n dygraph_loss: {}".format(static_loss,
+                                                             dygraph_loss))
+        self.verify_predict()
+
+    def test_in_static_mode_mkldnn(self):
+        paddle.fluid.set_flags({'FLAGS_use_mkldnn': True})
```
Comment (Aurelius84) [inline python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:354]: Mark `paddle.fluid.set_flags`

**Review by Aurelius84 — DISMISSED** [review]
LGTM. I just have marked some unsupported cases to help locate the code.

**Review by liym27 — DISMISSED** [review]
LGTM

---

All APIs in `fluid.XX` module need to be replaced

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py (hunk 1 above)
Comment (zhhsplendid) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Done

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py (hunk 2 above)
Comment (zhhsplendid) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py]: Done

File: python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py (hunk 3 above)
Comment (zhhsplendid) [reply python/paddle/fluid/tests/unittests/dygraph_to_static/test_resnet_v2.py:127]: We don't have to change here

**Review by Aurelius84 — APPROVED** [review]

**Review by liym27 — APPROVED** [review]
LGTM

**Pull request merged**